packages/js/product-editor/src/blocks/product-fields/product-details-section-description/edit.tsx

```diff
@@ -0,0 +1,229 @@
+import { useWooBlockProps } from '@woocommerce/block-templates';
+import { Product } from '@woocommerce/data';
+import { getNewPath, navigateTo } from '@woocommerce/navigation';
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore No types for this exist yet.
+// eslint-disable-next-line @woocommerce/dependency-group
+import { useEntityId } from '@wordpress/core-data';
+
+/**
+ * Internal dependencies
+ */
+import { ProductEditorSettings } from '../../../components';
+import { ProductTemplate } from '../../../components/editor';
+import { AUTO_DRAFT_NAME } from '../../../utils';
+import { ProductEditorBlockEditProps } from '../../../types';
+import { ProductDetailsSectionDescriptionBlockAttributes } from './types';
+
+export function ProductDetailsSectionDescriptionBlockEdit( {
+	attributes,
+	clientId,
+}: ProductEditorBlockEditProps< ProductDetailsSectionDescriptionBlockAttributes > ) {
+	const blockProps = useWooBlockProps( attributes );
+	const [ productType ] = useEntityProp( 'postType', 'product', 'type' );
+
+	const { productTemplates } = useSelect( ( select ) => {
+		const { getEditorSettings } = select( 'core/editor' );
+		return getEditorSettings() as ProductEditorSettings;
+	} );
+
+	const [ supportedProductTemplates, unsupportedProductTemplates ] =
+		productTemplates.reduce< [ ProductTemplate[], ProductTemplate[] ] >(
+			( [ supported, unsupported ], productTemplate ) => {
+				if ( productTemplate.layout_template_id ) {
+					supported.push( productTemplate );
+				} else {
+					unsupported.push( productTemplate );
+				}
+				return [ supported, unsupported ];
+			},
+			[ [], [] ]
+		);
+
+	const productId = useEntityId( 'postType', 'product' );
+
+	const hasEdits = useSelect(
+		( select ) => {
+			// @ts-expect-error There are no types for this.
+			const { hasEditsForEntityRecord } = select( 'core' );
+			return hasEditsForEntityRecord( 'postType', 'product', productId );
+		},
+		[ productId ]
+	);
+
+	const { saveEntityRecord } = useDispatch( 'core' );
+	const { createErrorNotice } = useDispatch( 'core/notices' );
+
+	const rootClientId = useSelect(
+		( select ) => {
+			const { getBlockRootClientId } = select( 'core/block-editor' );
+			return getBlockRootClientId( clientId );
+		},
+		[ clientId ]
+	);
+
+	if ( ! rootClientId ) return;
+
+	function menuItemClickHandler(
+		productTemplate: ProductTemplate,
+		onClose: () => void
+	) {
+		return async function handleMenuItemClick() {
+			if ( hasEdits ) {
+				createErrorNotice(
+					__(
+						'The current product must be saved before continue.',
+						'woocommerce'
+					)
+				);
+				return;
+			}
+
+			const newProduct = await ( saveEntityRecord(
+				'postType',
+				'product',
+				{
+					title: AUTO_DRAFT_NAME,
+					status: 'auto-draft',
+					...productTemplate.product_data,
+				}
+			) as never as Promise< Product > );
+
+			const url = getNewPath( {}, `/product/${ newProduct.id }` );
+			navigateTo( { url } );
+
+			onClose();
```

Comment:
Is this part still in-progress? 

- It doesn't auto-save the changes before switching
- It always created a new draft when switching, instead of changing the current product

Reply:
done